### DISCUSSION TITLE
how to run simulator or agent on ubuntu vbox?

### DISCUSSION THREAD

**Discussion by Sumitnaik22** [mtconnect/cppagent#309] [Q&A] [answered]
I am currently started working on Mtconnect , trying to understand it by implementinfg it on vbox.
I followed instruction given in github to build agent, was able to set up build ,download the source and build the agent using these command:

sudo apt-get install build-essential python3 python3-pip git cmake ruby rake autoconf
python3 -m pip install conan
echo 'export PATH=$HOME/.local/bin:$PATH' >> .bashrc

git clone https://github.com/mtconnect/cppagent.git

conan create cppagent -pr cppagent/conan/profile/gcc --build=missing

Now i am no able to figure out how to run simulator or run the agent.

Any guidance would be of great help.
Thank you.

**Comment by robot-ranger** [comment]
assuming it is all built correctly, ~i think the launch cmd `mtcagent run` should launch the agent as well as run the simulator~

once you have your own `agent.cfg` and `devices.xml`, you can run `mtcagent run /your/path/to/agent.cfg`

EDIT:

i havent built it outside a docker container in a while. see here: https://www.mtcup.org/Installing_C++_Agent_on_Ubuntu#run-the-agent. but one note: the exe has changed name from `agent` to `mtcagent`. you need to launch the ruby simulator script separately. launch it after you launch the agent with `mtcagent run agent.cfg` im sorry, i dont know exactly where the ruby scripts get installed

**Comment by wsobel** [comment, marked as answer]
Checkout the Dockerfile. It goes through the steps on a clean Ubuntu image. When you run the image, it starts the demo simulator. - W
